packages/ZbayLite/src/rtl-tests/channel.add.test.tsx

```diff
@@ -0,0 +1,133 @@
+
+    const factory = await getFactory(store)
+
+    await factory.create<ReturnType<typeof identity.actions.addNewIdentity>['payload']>(
+      'Identity',
+      { zbayNickname: 'holmes' }
+    )
+
+    renderComponent(
+      <>
+        <Sidebar />
+        <CreateChannel />
+      </>,
+      store
+    )
+
+    const addChannel = screen.getByTestId('addChannelButton')
+    userEvent.click(addChannel)
+
+    const title = await screen.findByText('Create a new public channel')
+    expect(title).toBeVisible()
+  })
+
+  it('Adds new channel and opens it', async () => {
+    const { store, runSaga } = await prepareStore(
+      {
+        [StoreKeys.Modals]: {
+          ...new ModalsInitialState(),
+          [ModalName.createChannel]: { open: true }
+        }
+      },
+      socket // Fork Nectar's sagas
+    )
+
+    const factory = await getFactory(store)
+
+    const holmes = await factory.create<
+    ReturnType<typeof identity.actions.addNewIdentity>['payload']
+    >('Identity', { zbayNickname: 'holmes' })
+
+    jest
+      .spyOn(socket, 'emit')
+      .mockImplementation(async (action: SocketActionTypes, ...input: any[]) => {
+        if (action === SocketActionTypes.SUBSCRIBE_FOR_TOPIC) {
+          const data = input as socketEventData<[string, PublicChannel]>
+          expect(data[0]).toEqual(holmes.peerId.id)
+          expect(data[1].name).toEqual('my-super-channel')
+          return socket.socketClient.emit(SocketActionTypes.CREATED_CHANNEL, {
+            channel: data[1],
+            communityId: holmes.id // Identity id is the same as community id
+          })
+        }
+      })
+
+    renderComponent(
+      <>
+        <CreateChannel />
+        <Channel />
+      </>,
+      store
+    )
+
+    const input = screen.getByPlaceholderText('Enter a channel name')
+    userEvent.type(input, 'my Super Channel')
+
+    // Check if parsed channel name displays properly
+    const info = screen.getByText('#my-super-channel')
+    expect(info).toBeVisible()
+
+    const button = screen.getByText('Create Channel')
+    userEvent.click(button)
+
+    await act(async () => {
+      await runSaga(testCreateChannelSaga).toPromise()
+    })
+
+    function* testCreateChannelSaga(): Generator {
+      const createChannelAction = yield* take(publicChannels.actions.createChannel)
+      expect(createChannelAction.payload.channel.name).toEqual('my-super-channel')
+      expect(createChannelAction.payload.channel.owner).toEqual(holmes.zbayNickname)
+      const addChannelAction = yield* take(publicChannels.actions.addChannel)
+      expect(addChannelAction.payload.channel).toEqual(createChannelAction.payload.channel)
+    }
+
+    const createChannelModal = screen.queryByTestId('createChannelModal')
+    expect(createChannelModal).toBeNull()
+
+    // Check if newly created channel is present and selected
+    const channelTitle = screen.getByTestId('channelTitle')
+    expect(channelTitle).toHaveTextContent('#my-super-channel')
```

Comment:
I think we can

Reply:
let's!

Reply:
The problematic thing is we use hooks for styling elements so the output contains generated class names like "makeStyles-selected-539". I added assertion for checking if sidebar item displays as selected in this scenario (by hardcoding the generated class name), but I don't want to repeat the same thing for the other tests (channel.switch.test) because I'm not sure if it won't break later.